test/parallel/test-http2-client-socket-operations.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const h2 = require('http2');
+
+const server = h2.createServer();
+server.on('stream', common.mustCall((stream) => {
+  stream.respond();
+  stream.end();
+}));
+
+server.listen(0, common.mustCall(() => {
+  const client = h2.connect(`http://localhost:${server.address().port}`);
+  const req = client.request({ ':path': '/' });
+
+  req.on(
+    'response',
+    common.mustCall(() => {
+      client.socket.pause();
+      // Force 'pause' when _handle.reading is false
+      client.socket.emit('pause');
```

Comment:
Reposting this since github incorrectly hides it.

> I'll leave this up to @jasnell & @mcollina but I don't think this is actually testing what it claims to. Feels like a way to bump up code coverage without truly testing the functionality which is worse than not having a test since it hides potential problem areas.
> 
> Unfortunately I don't really have great suggestion for how to properly test it given how the socket is used in h2.

Reply:
I'd tried mocking calls to `readStart` and `readStop` the same way as `submitSettings` is mocked as discussed in https://github.com/nodejs/node/issues/16083#issuecomment-335022398

However, the call was not reaching the mock. I'll wait for inputs from @jasnell / @mcollina 
I agree that current commit doesn't confirm that pause, resume and drain methods were called.

Reply:
One solution I can think of is ensuring that operations were called on `client.socket` as follows:
```
client.socket.on('pause', common.mustCall(2));
client.socket.on('resume', common.mustCall(2));
client.socket.on('drain', common.mustCall());
```